airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcCtidInitializer.java

```diff
@@ -0,0 +1,181 @@
+                : ctidStreams.streamsForCtidSync().stream()
+                    .filter(c -> !streamsUnderVacuum.contains(AirbyteStreamNameNamespacePair.fromConfiguredAirbyteSteam(c)))
+                    .toList();
+        LOGGER.info("Streams to be synced via ctid : {}", finalListOfStreamsToBeSyncedViaCtid.size());
+        final Map<io.airbyte.protocol.models.AirbyteStreamNameNamespacePair, Long> fileNodes = PostgresQueryUtils.fileNodeForStreams(database,
+            finalListOfStreamsToBeSyncedViaCtid,
+            quoteString);
+        final CtidStateManager ctidStateManager = new CtidGlobalStateManager(ctidStreams, fileNodes, stateToBeUsed, catalog);
+        final CtidPostgresSourceOperations ctidPostgresSourceOperations = new CtidPostgresSourceOperations(
+            Optional.of(new CdcMetadataInjector(
+                emittedAt.toString(), io.airbyte.db.PostgresUtils.getLsn(database).asLong(), new PostgresCdcConnectorMetadataInjector())));
+        final Map<io.airbyte.protocol.models.AirbyteStreamNameNamespacePair, TableBlockSize> tableBlockSizes =
+            PostgresQueryUtils.getTableBlockSizeForStream(
+                database,
+                finalListOfStreamsToBeSyncedViaCtid,
+                quoteString);
+        final PostgresCtidHandler ctidHandler = new PostgresCtidHandler(sourceConfig, database,
+            ctidPostgresSourceOperations,
+            quoteString,
+            fileNodes,
+            tableBlockSizes,
+            ctidStateManager,
+            namespacePair -> Jsons.emptyObject());
+
+        ctidIterator.addAll(ctidHandler.getIncrementalIterators(
+            new ConfiguredAirbyteCatalog().withStreams(finalListOfStreamsToBeSyncedViaCtid), tableNameToTable, emittedAt));
+      } else {
+        LOGGER.info("No streams will be synced via ctid");
+      }
+
+      final AirbyteDebeziumHandler<Long> handler = new AirbyteDebeziumHandler<>(sourceConfig,
+          PostgresCdcTargetPosition.targetPosition(database), false, firstRecordWaitTime, queueSize);
+      final PostgresCdcStateHandler postgresCdcStateHandler = new PostgresCdcStateHandler(stateManager);
+
+      final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorSupplier = () -> handler.getIncrementalIterators(catalog,
+          new PostgresCdcSavedInfoFetcher(stateToBeUsed),
+          postgresCdcStateHandler,
+          new PostgresCdcConnectorMetadataInjector(),
+          PostgresCdcProperties.getDebeziumDefaultProperties(database),
+          emittedAt,
+          false);
+
+      if (ctidIterator.isEmpty()) {
+        return Collections.singletonList(incrementalIteratorSupplier.get());
+      }
+
+      if (streamsUnderVacuum.isEmpty()) {
+        // This starts processing the WAL as soon as initial sync is complete, this is a bit different from the current cdc syncs.
+        // We finish the current CDC once the initial snapshot is complete and the next sync starts processing the WAL
+        return Stream
```

Comment:
Interesting. 
So if I'm getting it, streams yet to be fully synced by ctid will show up twice - once as ctid, and the other as WAL through debezium